frontend/server/src/Controllers/Course.php

```diff
@@ -591,9 +591,27 @@ public static function apiClone(\OmegaUp\Request $r): array {
                 }
             }
             \OmegaUp\DAO\DAO::transEnd();
+            $result = 'success';
         } catch (\Exception $e) {
             \OmegaUp\DAO\DAO::transRollback();
+            $result = 'unknown';
             throw $e;
+        } finally {
+            $courseId = null;
+            if (isset($course) && !is_null($course->course_id)) {
```

Comment:
```suggestion
            if (isset($course)) {
```

si es nulo, es un no-op. así incluso podríamos darle inline a esto como un ternario en L608.

Reply:
Listo